bliss/surveys/dc2.py

```diff
@@ -293,6 +186,160 @@ def test_dataloader(self):
             self.test_dataset, batch_size=self.batch_size, num_workers=self.num_workers // 2
         )
 
+    def get_plotting_sample(self, image_idx):
+        if self.image_files is None or self.bg_files is None:
+            self._load_image_and_bg_files_list()
+
+        (
+            image,
+            bg,
+            tile_dict,
+            full_cat,
+            wcs,
+            _,
+            match_id,
+            psf_params,
+        ) = load_image_and_catalog(
+            image_idx,
+            self.image_files,
+            self.bg_files,
+            self.image_lim,
+            self.bands,
+            self.n_bands,
+            self.cat_path,
+            self.min_flux_threshold,
+        )
+        return {
+            "tile_catalog": tile_dict,
+            "image": torch.from_numpy(image),
+            "background": torch.from_numpy(bg),
+            "match_id": match_id,
+            "full_catalog": full_cat,
+            "wcs": wcs,
+            "psf_params": psf_params,
+        }
+
+
+def squeeze_tile_dict(tile_dict):
+    tile_dict_copy = copy.copy(tile_dict)
+    for k, v in tile_dict_copy.items():
+        if k != "n_sources":
+            tile_dict_copy[k] = rearrange(v, "1 h w nh nw -> h w nh nw")
+    tile_dict_copy["n_sources"] = rearrange(tile_dict_copy["n_sources"], "1 h w -> h w")
+    return tile_dict_copy
+
+
+def unsqueeze_tile_dict(tile_dict):
+    tile_dict_copy = copy.copy(tile_dict)
+    for k, v in tile_dict_copy.items():
+        if k != "n_sources":
+            tile_dict_copy[k] = rearrange(v, "h w nh nw -> 1 h w nh nw")
+    tile_dict_copy["n_sources"] = rearrange(tile_dict_copy["n_sources"], "h w -> 1 h w")
+    return tile_dict_copy
+
+
+def load_image_and_catalog(
+    image_idx, image_files, bg_files, image_lim, bands, n_bands, cat_path, min_flux_threshold
+):
+    image_list, bg_list, wcs_header_str = read_frame_for_band(
+        image_files, bg_files, image_idx, n_bands, image_lim
+    )
+    wcs = from_wcs_header_str_to_wcs(wcs_header_str)
+    image = np.stack(image_list)
+    bg = np.stack(bg_list)
+
+    plocs_lim = image[0].shape
+    height = plocs_lim[0]
+    width = plocs_lim[1]
+    full_cat, psf_params, match_id = Dc2FullCatalog.from_file(
+        cat_path, wcs, height, width, bands, min_flux_threshold
+    )
+    tile_dict = full_cat.to_tile_catalog(4, 5).get_brightest_sources_per_tile().to_dict()
+    tile_dict = squeeze_tile_dict(tile_dict)
+    tile_dict["star_fluxes"] = tile_dict["star_fluxes"].clamp(min=1e-18)
+    tile_dict["galaxy_fluxes"] = tile_dict["galaxy_fluxes"].clamp(min=1e-18)
+    tile_dict["galaxy_params"][..., 3] = tile_dict["galaxy_params"][..., 3].clamp(min=1e-18)
+    tile_dict["galaxy_params"][..., 5] = tile_dict["galaxy_params"][..., 5].clamp(min=1e-18)
+
+    return image, bg, tile_dict, full_cat, wcs, wcs_header_str, match_id, psf_params  # noqa: WPS227
```

Comment:
Using a dict is one way to bundle together parameters, ideally into meaningful groups, rather than just putting them all together to evade the flake8 check

Reply:
have fixed it